mlir/include/mlir/Dialect/Rock/utility/loweringUtils.h

```diff
@@ -22,6 +22,76 @@ namespace rock {
 struct ConvolutionDims;
 struct GemmSize;
 
+// This is a helper data structure
+// to server as a return type for
+// view creating utilities.
+struct GPUViews {
+  // This is a [gridIdx0, ... , gridIdxN, tid, iter] to K x D view
+  // This is useful to define a global tile once loaded into regs.
+  ArrayAttr gridwiseView;
+  // This is a [tid, iter] to flat K x D view
+  // Useful to define a blockwise tile once loaded into regs.
+  ArrayAttr blockwiseView;
+  // This is a [iter] to flat K x D view
+  // Useful to define a threadwise tile once loaded into regs.
+  ArrayAttr threadwiseView;
+};
+
+// This is helper data structure to allow
+// lowering define lower dimensions (K or D) re-shuffling
+// into thread_id or iter into reg buffers in threads.
+// It is particularly useful to arrange data
+// in registers before they are loaded/written
+// to some memory (global or lds).
+struct LowerDimPartInfo {
+  // This defines sub-dimension names of either tid or iter.
+  // These sub-dimensions will be part of the merger when creating
+  // the tid or iter dimensions.
+  SmallVector<StringRef> lowerDimPartNames;
```

Comment:
Returning a `StringRef` is extremely suspicious on memory safety grounds.

Reply:
(It'll probably work, but I'd be careful here if anyone is constructing derived strings from this value. If nothing else, warn about it)